satpy/readers/mirs.py

```diff
@@ -0,0 +1,502 @@
+
+        if len(self.nc.coords.values()) == 0:
+            self.nc = self.nc.assign_coords(self.new_coords())
+
+        self.platform_name = self._get_platform_name
+        self.sensor = self._get_sensor
+        self.lons = None
+        self.lats = None
+        self.coords = {}
+
+    def new_coords(self):
+        """Define coordinates when file does not use variable attributes."""
+        if not self.nc.coords.keys():
+            # this file did not define variable coordinates
+            new_coords = {'latitude': self['latitude'],
+                          'longitude': self['longitude']}
+        else:
+            # let xarray handle coordinates defined by variable attributes.
+            new_coords = self.nc.coords.keys()
+        return new_coords
+
+    @property
+    def platform_shortname(self):
+        """Get platform shortname."""
+        return self.filename_info['platform_shortname']
+
+    @property
+    def _get_platform_name(self):
+        """Get platform name."""
+        try:
+            res = PLATFORMS[self.filename_info['platform_shortname'].lower()]
+        except KeyError:
+            res = "mirs"
+        return res.lower()
+
+    @property
+    def _get_sensor(self):
+        """Get sensor."""
+        try:
+            res = SENSOR[self.filename_info["platform_shortname"].lower()]
+        except KeyError:
+            res = self.sensor_names
+        return res
+
+    @property
+    def sensor_names(self):
+        """Return standard sensor names for the file's data."""
+        return list(set(SENSOR.values()))
+
+    @property
+    def start_time(self):
+        """Get start time."""
+        # old file format
+        if self.filename_info.get("date", False):
+            s_time = datetime.datetime.combine(
+                self.force_date("date"),
+                self.force_time("start_time")
+            )
+            self.filename_info["start_time"] = s_time
+        return self.filename_info["start_time"]
+
+    @property
+    def end_time(self):
+        """Get end time."""
+        # old file format
+        if self.filename_info.get("date", False):
+            end_time = datetime.datetime.combine(
+                self.force_date("date"),
+                self.force_time("end_time")
+            )
+            self.filename_info["end_time"] = end_time
+        return self.filename_info["end_time"]
+
+    def force_date(self, key):
+        """Force datetime.date for combine."""
+        if isinstance(self.filename_info[key], datetime.datetime):
+            return self.filename_info[key].date()
+        else:
+            return self.filename_info[key]
+
+    def force_time(self, key):
+        """Force datetime.time for combine."""
+        if isinstance(self.filename_info.get(key), datetime.datetime):
+            return self.filename_info.get(key).time()
+        else:
+            return self.filename_info.get(key)
+
+    def limb_correct_atms_bt(self, bt_data, ds_info):
```

Comment:
Not exactly.  I started pulling the lines that use self out of this subroutine and it created an incredibly long function call.  If that is preferred, I can still make the change.   The way the coefficient filenames are stored would have to change, but I am not sure if they necessarily have to be part of the dask Dataset attributes.

Reply:
I'm not sure I understand. I mean that this method `limb_correct_atms_bt` is an instance method and has its first argument as `self` however it never actually uses `self` inside the method (from what I can tell). This means it could be made a `@staticmethod` or the method could be moved outside of the class to be a function.

Reply:
The method itself doesn't need it, you are correct.  I'll fix it.